solutions/tests/test_count_words.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""A module containing tests for the count_words function in solutions/count_words.py
+
+Module contents:
+    TestCountWords: A class that contains tests for the count_words function.
+
+Created on Sunday, 29th December, 2024.
+@author: Gai Samuel
+"""
+
+import unittest
+
+from solutions.count_words import count_words
+
+
+class TestCountWords(unittest.TestCase):
+    """Tests for count_words function"""
+
+    def test_count_words(self):
+        """count_words returns the correct word count for each word in the text"""
+        self.assertEqual(
+            count_words("My name is Gai"), {"my": 1, "name": 1, "is": 1, "gai": 1}
+        )
+
+    def test_count_words_with_an_empty_string(self):
+        """count_words returns an empty dictionary for an empty string"""
+        self.assertEqual(count_words(""), {})
+
+    def test_count_words_with_punctuation(self):
+        """Throws an assertion error if the punctuation is declared in it's own string"""
+        self.assertEqual(
+            count_words("My name is Gai."), {"my": 1, "name": 1, "is": 1, "gai.": 1}
+        )
+
+    def test_count_words_with_uppercase(self):
+        """Throws an assertion error if the text in output string is in uppercase"""
+        self.assertEqual(
+            count_words("MY NAME IS GAI"), {"my": 1, "name": 1, "is": 1, "gai": 1}
+        )
+
+    def test_count_words_with_multiple_spaces(self):
+        """Throws an assertion error if there are multiple spaces in the text"""
+        self.assertEqual(
+            count_words("My  name  is  Gai"), {"my": 1, "name": 1, "is": 1, "gai": 1}
+        )
+
+    def test_count_words_differ_by_case(self):
+        """counts words that are the same but differ by case as same word"""
+        self.assertEqual(
+            count_words("GAI is good. Gai likes to code"),
+            {"gai": 2, "is": 1, "good.": 1, "likes": 1, "to": 1, "code": 1},
+        )
+
+    def test_count_words_with_numbers(self):
+        """counts the number in the text as a word of it's own"""
+        self.assertEqual(
+            count_words("Gai is 24 years old"),
+            {"gai": 1, "is": 1, "24": 1, "years": 1, "old": 1},
+        )
+
+    def test_count_words_with_special_characters(self):
+        """counts words with special characters and the special characters as same word"""
```

Comment:
for this does it mean that if the string was  : "Hello, Gai! How are you Gai?" 
it should count Gai! and Gai? as different strings ?

Reply:
I just tested the theory typed, and hello, Gai! is counted as one word, but if the comma is removed, then gai! and gai? are counted as different words. Now, to the above test, I was trying to see if the word and the special character are counted separately.